### PR TITLE
SoftMaker.FreePDF.2019 version 2.10

### PR DESCRIPTION
**Pull request by borque**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/89292)

### CODE REVIEW

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
Is there a need for having year in the PackageIdentifier? I'm seeing the package website https://www.getfreepdf.com/ and I don't see the publisher differentiating versions with release years.

**Comment by borque** [comment]
When you request a (free) registration for this current FreePDF the mail speaks explizitly of FreePDF 2019 after there was also a previous version called FreePDF 2017.
Furthermore looking at the paid version, FlexiPDF, which has versions 2017, 2019, 2022 and other SoftMaker products like FreeOffice and Office, which also carry a year in their versioning schema it is very likely a future version of FreePDF will be called 2022 or something like that.
So, though I can't say for sure it might be the more future proof way to call ist FreePDF.2019 in the PackageIdentifier as suggested?!

**Comment by mdanish-kh** [comment]
@borque Thanks for the reply! I understand your point that it may be safe to have year in the PackageIdentifier as is the case with other SoftMaker apps. But looking at the [version history](https://www.getfreepdf.com/en/service/version-history) and the fact that the year isn't mentioned in the ARP Entries as well (which is not the case with other SoftMaker Apps like FlexiPDF, [FreeOffice](https://github.com/microsoft/winget-pkgs/blob/16c2b8fcb520f7486a13e0745c7fb80826e4de75/manifests/s/SoftMaker/FreeOffice/2021/1.0.5290/SoftMaker.FreeOffice.2021.locale.en-US.yaml#L11), [Office](https://github.com/microsoft/winget-pkgs/blob/16c2b8fcb520f7486a13e0745c7fb80826e4de75/manifests/s/SoftMaker/Office/2021/21.0.5294/SoftMaker.Office.2021.locale.en-US.yaml#L11) that have year in their DisplayName), I don't see a convincing reason to believe that the package needs / or may need separation by year in the future. Even if it needs it, owing to the fact that the link is not version specific and only the latest version will be kept in the repository, the PackageIdentifier can be easily shifted. For now, the PackageIdentifier may cause unnecessary ambiguity for the user. I may be entirely wrong so I need more views on this.

@russellbanks - Do you have an opinion about it?

**Comment by russellbanks** [comment]
Following on from what @mdanish-kh has said, I would highlight this point:
> the year isn't mentioned in the ARP Entries as well (which is not the case with other SoftMaker Apps like FlexiPDF, [FreeOffice](https://github.com/microsoft/winget-pkgs/blob/16c2b8fcb520f7486a13e0745c7fb80826e4de75/manifests/s/SoftMaker/FreeOffice/2021/1.0.5290/SoftMaker.FreeOffice.2021.locale.en-US.yaml#L11), [Office](https://github.com/microsoft/winget-pkgs/blob/16c2b8fcb520f7486a13e0745c7fb80826e4de75/manifests/s/SoftMaker/Office/2021/21.0.5294/SoftMaker.Office.2021.locale.en-US.yaml#L11) that have year in their DisplayName)

I think this indicates the publisher's intention, so I agree that there would be no need here to have the year in the `PackageIdentifier`. If they ever changed that, then we can always add it in the future.

File: manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: SoftMaker.FreePDF.2019
+PackageVersion: 2.10
+PackageLocale: en-US
+Publisher: SoftMaker Software GmbH
+PackageName: FreePDF 2019
```
Comment (mdanish-kh) [inline manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.locale.en-US.yaml:8]:
```suggestion
PackageName: FreePDF
```
Must match Control Panel / Apps & Features.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Comment by borque** [comment]
You both have a good points here. I have just accepted the changes suggested by @mdanish-kh and the verification-pipelines started working again. 
Is this manifest in an acceptable shape after that, or should I better close this PR and resubmit the whole thing with the requested changes as SoftMaker.FreePDF (without the year)?

File: manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: SoftMaker.FreePDF.2019
```
Comment (mdanish-kh) [inline manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.installer.yaml:4]:
```suggestion
PackageIdentifier: SoftMaker.FreePDF
```

File: manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: SoftMaker.FreePDF.2019
```
Comment (mdanish-kh) [inline manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.locale.en-US.yaml:4]:
```suggestion
PackageIdentifier: SoftMaker.FreePDF
```

File: manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.2.0.schema.json
+
+PackageIdentifier: SoftMaker.FreePDF.2019
```
Comment (mdanish-kh) [inline manifests/s/SoftMaker/FreePDF/2019/2.10/SoftMaker.FreePDF.2019.yaml:4]:
```suggestion
PackageIdentifier: SoftMaker.FreePDF
```

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Comment by mdanish-kh** [comment]
After making the above changes, you would also have to modify the directory structure to meet the new PackageIdentifier. You could do this in number of ways through Git / GitHub. A neat way of doing it is here in GitHub Web by pressing <kbd>.</kbd> (dot) key on your keyboard that will open up VSCode Online to make changes to this PR.

- Navigate to Explorer menu on the left and find your modified files
- Move the `2.10` folder from `manifests/s/SoftMaker/FreePDF/2019/2.10` to `manifests/s/SoftMaker/FreePDF/2.10` and delete the `2019` folder.
- Stage all changes and commit from the Source Control menu on the left. (the changes are automatically pushed)

The directories should look similar to this:

**Comment by borque** [comment]
Hello @mdanish-kh!
A great Thank you for teaching me this way how to use Github this way! Thats great. 
Hopefully I succeeded with my latest try to change the commit?!

**Comment by mdanish-kh** [comment]
> 



> Hello @mdanish-kh! A great Thank you for giving me a helping hand and teaching me this way how to use Github! Thats great. Hopefully I succeeded with my latest try to change the commit?!

My pleasure :) The changes look good to me now.

**Review by mdanish-kh — APPROVED** [review]

**Pull request merged**